src/hints/squashedDictManager.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  EmptyIndex,
+  EmptyIndices,
+  EmptyKeys,
+} from 'errors/squashedDictManager';
+
+import { Felt } from 'primitives/felt';
+
+/**
+ *  Handle the squashing of dictionnaries.
+ */
+export class SquashedDictManager {
+  /** Maps the key of a dictionnary to its taken values accross the run. */
+  public keyToIndices: Map<string, Felt[]>;
+  /** An array containing the keys that still needs to be squashed. */
+  public keys: Felt[];
+
+  constructor() {
+    this.keyToIndices = new Map<string, Felt[]>();
+    this.keys = [];
+  }
+
+  /** Add `index` to the indices taken by `key` */
+  insert(key: Felt, index: Felt) {
+    const keyStr = key.toString();
+    const indices = this.keyToIndices.get(keyStr);
+    if (!indices) {
+      this.keyToIndices.set(keyStr, [index]);
+    } else {
+      indices.push(index);
+    }
+  }
```

Comment:
maybe we can consider using a `DefaultMap` as per examples here https://stackoverflow.com/questions/51319147/map-default-value instead